components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/AccessTokenIssuer.java

```diff
@@ -782,9 +786,36 @@ private String getSubjectClaimFromUserStore(String subjectClaimUri, Authenticate
         AbstractUserStoreManager userStoreManager = (AbstractUserStoreManager) IdentityTenantUtil
                 .getRealm(authenticatedUser.getTenantDomain(), authenticatedUser.toFullQualifiedUsername())
                 .getUserStoreManager();
+        if (userStoreManager.isExistingUserWithID(authenticatedUser.getUserId())) {
+            return userStoreManager
+                    .getUserClaimValueWithID(authenticatedUser.getUserId(), subjectClaimUri, null);
+        }
 
-        return userStoreManager
-                .getUserClaimValueWithID(authenticatedUser.getUserId(), subjectClaimUri, null);
+        /*
+            If the user is not in the user store, check if the user can be resolved from ancestor organizations and
+            resolve the subject claim.
+        */
+        try {
```

Comment:
invoke this only if organization management enabled

Reply:
let's extract this to a private method too

Reply:
Fixed by - https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2103/commits/b3a1b2a1838130a38dc197f0ab9658d22237ab37